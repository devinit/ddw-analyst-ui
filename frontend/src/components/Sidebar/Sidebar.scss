@media (max-width: 991px) {
  .off-canvas-sidebar nav .navbar-collapse:before, .sidebar:before {
    background-color: inherit;
  }
}
