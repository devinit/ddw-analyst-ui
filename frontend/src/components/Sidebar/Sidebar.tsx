import * as React from 'react';
import { SidebarContent } from '../SidebarContent';
import { SidebarItem } from '../SidebarItem';
import { SidebarLink } from '../SidebarLink';
import { SidebarLogo } from '../SidebarLogo';

interface SidebarProps {
  dataColour?: 'rose' | 'purple' | 'azure' | 'danger' | 'green' | 'orange';
  backgroundColour?: 'black' | 'white' | 'red';
}

export class Sidebar extends React.Component<SidebarProps> {
  static defaultProps: Partial<SidebarProps> = {
    dataColour: 'rose',
    backgroundColour: 'black'
  };
  static Logo = SidebarLogo;
  static Content = SidebarContent;
  static Item = SidebarItem;
  static Link = SidebarLink;

  render() {
    return (
      <div
        className="sidebar"
        data-color={ this.props.dataColour }
        data-background-color={ this.props.backgroundColour }
      >
        { this.renderContent(SidebarLogo) }
        { this.renderContent(SidebarContent) }
      </div>
    );
  }

  private renderContent(type: typeof SidebarLogo | typeof SidebarContent) {
    return React.Children.map(this.props.children, child => {
      if (React.isValidElement(child) && child.type === type) {
        return child;
      }
    });
  }
}
