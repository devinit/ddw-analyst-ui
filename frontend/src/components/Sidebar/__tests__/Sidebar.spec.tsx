import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Sidebar } from '../Sidebar';

test('renders correctly with the default props', () => {
  const renderer = TestRenderer.create(<Sidebar />).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders with the specified dataColour', () => {
  const azureRenderer = TestRenderer.create(<Sidebar dataColour="azure" />).toJSON();

  expect(azureRenderer).toMatchSnapshot();

  const dangerRenderer = TestRenderer.create(<Sidebar dataColour="danger" />).toJSON();

  expect(dangerRenderer).toMatchSnapshot();

  const greenRenderer = TestRenderer.create(<Sidebar dataColour="green" />).toJSON();

  expect(greenRenderer).toMatchSnapshot();

  const orangeRenderer = TestRenderer.create(<Sidebar dataColour="orange" />).toJSON();

  expect(orangeRenderer).toMatchSnapshot();

  const purpleRenderer = TestRenderer.create(<Sidebar dataColour="purple" />).toJSON();

  expect(purpleRenderer).toMatchSnapshot();

  const roseRenderer = TestRenderer.create(<Sidebar dataColour="rose" />).toJSON();

  expect(roseRenderer).toMatchSnapshot();
});

test('renders with the specified backgroundColour', () => {
  const blackRenderer = TestRenderer.create(<Sidebar backgroundColour="black" />).toJSON();

  expect(blackRenderer).toMatchSnapshot();

  const redRenderer = TestRenderer.create(<Sidebar backgroundColour="red" />).toJSON();

  expect(redRenderer).toMatchSnapshot();

  const whiteRenderer = TestRenderer.create(<Sidebar backgroundColour="white" />).toJSON();

  expect(whiteRenderer).toMatchSnapshot();
});

test('does not render invalid children', () => {
  const renderer = TestRenderer.create(
    <Sidebar>
      <div>Awesome</div>
    </Sidebar>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders a SidebarLogo when one is added as a child', () => {
  const renderer = TestRenderer.create(
    <Sidebar>
      <Sidebar.Logo />
    </Sidebar>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders a SidebarContent when one is added as a child', () => {
  const renderer = TestRenderer.create(
    <Sidebar>
      <Sidebar.Content />
    </Sidebar>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});
