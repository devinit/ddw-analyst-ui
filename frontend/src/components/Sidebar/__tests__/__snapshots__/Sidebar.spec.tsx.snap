// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render invalid children 1`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="rose"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders a SidebarContent when one is added as a child 1`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="rose"
>
  <div
    className="sidebar-wrapper"
  >
    <ul
      className="nav"
    />
  </div>
</div>
`;

exports[`renders a SidebarLogo when one is added as a child 1`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="rose"
>
  <div
    className="logo"
  />
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders correctly with the default props 1`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="rose"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified backgroundColour 1`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="rose"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified backgroundColour 2`] = `
<div
  className="sidebar"
  data-background-color="red"
  data-color="rose"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified backgroundColour 3`] = `
<div
  className="sidebar"
  data-background-color="white"
  data-color="rose"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified dataColour 1`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="azure"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified dataColour 2`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="danger"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified dataColour 3`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="green"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified dataColour 4`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="orange"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified dataColour 5`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="purple"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;

exports[`renders with the specified dataColour 6`] = `
<div
  className="sidebar"
  data-background-color="black"
  data-color="rose"
>
  <div
    className="sidebar-wrapper"
  />
</div>
`;
